tests/test_paperqa.py

```diff
@@ -1199,3 +1199,68 @@ def test_answer_rename(recwarn) -> None:
 def test_dois_resolve_to_correct_journals(doi_journals):
     details = DocDetails(doi=doi_journals["doi"])  # type: ignore[call-arg]
     assert details.journal == doi_journals["journal"]
+
+
+@pytest.mark.vcr
+@pytest.mark.parametrize("use_partition", [True, False])
+@pytest.mark.asyncio
+async def test_partitioning_fn_docs(use_partition: bool) -> None:
+    settings = Settings.from_name("fast")
+    settings.answer.evidence_k = 2  # limit to only 2
+
+    # imagine we have some special selection we want to
+    # embedding rank by itself
+    def partition_by_journal(doc: Doc) -> int:
+        if "special" in doc.citation:
+            return 1
+        return 0
+
+    partitioning_fn = None
+
+    if use_partition:
+        partitioning_fn = partition_by_journal
+
+    docs = Docs(partitioning_fn=partitioning_fn)
+
+    assert isinstance(
+        docs.texts_index, NumpyVectorStore
+    ), "We want this test to cover NumpyVectorStore"
+
+    # add docs that we can use our partitioning function on
+    stub_doc = Doc(docname="stub", citation="stub", dockey="stub")
+    special_doc = Doc(docname="special", citation="special", dockey="special")
```

Comment:
Can you name these better? Something like:

```python
positive_statements_doc = Doc(...)
negative_statements_doc = Doc(...)
```

I think this contrast makes the test easier to read than "stub" vs "special"